x-pack/plugins/security_solution/public/entity_analytics/components/asset_criticality_selector.tsx

```diff
@@ -0,0 +1,233 @@
+
+import type { UseMutationResult, UseQueryResult } from '@tanstack/react-query';
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+
+import React, { useState } from 'react';
+import { useToggle } from 'react-use';
+import {
+  ASSET_CRITICALITY_OPTION_TEXT,
+  CREATE_ASSET_CRITICALITY,
+  CRITICALITY_LEVEL_COLOR,
+  PICK_ASSET_CRITICALITY,
+} from '../../../common/asset_criticality';
+import type { EntityAnalyticsPrivileges } from '../../../common/api/entity_analytics/common';
+import type { AssetCriticalityRecord } from '../../../common/api/entity_analytics/asset_criticality';
+
+import {
+  createAssetCriticality,
+  fetchAssetCriticality,
+  fetchAssetCriticalityPrivileges,
+} from '../api/api';
+
+interface Props {
+  entity: Entity;
+}
+export const AssetCriticalitySelector: React.FC<Props> = ({ entity }) => {
+  const criticality = useAssetCriticality(entity);
+
+  if (criticality.privileges.isLoading || !criticality.privileges.data?.has_all_required) {
+    return null;
+  }
+
+  return (
+    <>
+      <EuiAccordion
+        id="asset-criticality-selector"
+        buttonContent="Asset Criticality"
+        data-test-subj="asset-criticality-selector"
+      >
+        {criticality.query.isLoading || criticality.mutation.isLoading ? (
+          <EuiLoadingSpinner size="s" />
+        ) : (
+          <EuiFlexGroup direction="row" alignItems="center" wrap={false}>
+            <EuiFlexItem>
+              <EuiText size="s">
+                {criticality.status === 'update' && criticality.query.data?.criticality_level ? (
+                  <EuiHealth
+                    color={CRITICALITY_LEVEL_COLOR[criticality.query.data.criticality_level]}
+                  >
+                    {criticalityDisplayText[criticality.query.data.criticality_level]}
+                  </EuiHealth>
+                ) : (
+                  <EuiHealth color="subdued">{CREATE_ASSET_CRITICALITY}</EuiHealth>
+                )}
+              </EuiText>
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiButtonEmpty
+                data-test-subj="asset-criticality-change-btn"
+                iconType="arrowStart"
+                iconSide="left"
+                flush="right"
+                onClick={criticality.modal.toggle(true)}
+              >
+                {criticality.status === 'update' ? 'Change' : 'Create'}
+              </EuiButtonEmpty>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        )}
+      </EuiAccordion>
+      {criticality.modal.visible ? (
+        <AssetCriticalityModal entity={entity} criticality={criticality} />
+      ) : null}
+    </>
+  );
+};
+
+interface ModalProps {
+  criticality: State;
+  entity: Entity;
+}
+const AssetCriticalityModal: React.FC<ModalProps> = ({ criticality, entity }) => {
+  const basicSelectId = useGeneratedHtmlId({ prefix: 'basicSelect' });
+  const [value, setNewValue] = useState<AssetCriticalityRecord['criticality_level']>(
```

Comment:
TIP: We can export an `AssetCriticalityLevel` type, so we don't have to write `AssetCriticalityRecord['criticality_level']` everywhere, and it's much cleaner.